### ISSUE TITLE
CommonLib library

### ISSUE THREAD

**Issue by ggajoch** [cparsell/music-spider#1]
Hi!

First of all, thanks for that piece of software, that looks like exactly the thing I was looking for.

I do, however have one safety concern - I see that you are using a CommonLib library, which I cannot find the sources for - and due to security reasons I will not grant full access to my account for software which is not in my control.

Maybe it would be posible to include CommonLib in this source code too?

On the side note - I see that the app requires full access to GMail account, even for deleting every email. Is that necessary?

Best regards!
Grzegorz Gajoch

**Comment by cparsell** [comment]
Ah, CommonLibrary is a library I made a bit more recently to centralize a lot of the redundant functions I use across multiple Google Apps Script projects. I'll try to make it public on GitHub and include the ID for adding it. Thanks for pointing it out.

**Comment by cparsell** [comment]
I made the CommonLibrary repo and its Apps Script project public. I've linked to it in the Readme.

Note that in the Music-Spider project, I shorten CommonLibrary to CommonLib when linking it. 

Let me know if you see any issues. No one had really used this yet so this was an oversight on my part.

**Comment by cparsell** [comment]
Oh and regarding your question about GMail access, it should not require that level of access. At the time I wrote it, I found the scope that seemed to work for me but that level of access is definitely too broad. It only needs scope for sending an email message that lists the events it has found.

I'll try to look into a more limited scope that serves this function. If you're aware of one that would work let me know.
